role controller panics when the secret is not found

I try to apply the sample `dataplane_v1beta1_openstackdataplanerole.yaml` which has no ssh secret defined, it resulted in a panic:

```
2023-03-17T10:57:14Z	INFO	Observed a panic in reconciler: runtime error: invalid memory address or nil pointer dereference	{"controller": "openstackdataplanerole", "controllerGroup": "dataplane.openstack.org", "controllerKind": "OpenStackDataPlaneRole", "OpenStackDataPlaneRole": {"name":"openstackdataplanerole-sample","namespace":"openstack"}, "namespace": "openstack", "name": "openstackdataplanerole-sample", "reconcileID": "cf107eb6-f69b-4efe-8bde-5327b6b1c3cc"}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x15de837]

goroutine 479 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:119 +0x1fa
panic({0x17a2320, 0x287a120})
	/usr/local/go/src/runtime/panic.go:884 +0x212
github.com/openstack-k8s-operators/dataplane-operator/controllers.(*OpenStackDataPlaneRoleReconciler).Reconcile.func1()
	/remote-source/controllers/openstackdataplanerole_controller.go:147 +0x177
github.com/openstack-k8s-operators/dataplane-operator/controllers.(*OpenStackDataPlaneRoleReconciler).Reconcile(0xc0007a9200, {0x1c3ccd8, 0xc000804660}, {{{0xc0007e3e30?, 0x10?}, {0xc000500640?, 0x40da67?}}})
	/remote-source/controllers/openstackdataplanerole_controller.go:173 +0x1502
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x1c3ccd8?, {0x1c3ccd8?, 0xc000804660?}, {{{0xc0007e3e30?, 0x16dbc00?}, {0xc000500640?, 0x0?}}})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:122 +0xc8
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc00047ea00, {0x1c3cc30, 0xc0007a8140}, {0x1816b40?, 0xc000684140?})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:323 +0x38f
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc00047ea00, {0x1c3cc30, 0xc0007a8140})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:274 +0x1d9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:235 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:231 +0x333
```

I think it is related to the stauts.condition handling during the defer function but I did not looked at it deeply.

--- a/controllers/openstackdataplanenode_controller.go
+++ b/controllers/openstackdataplanenode_controller.go
@@ -139,14 +139,12 @@ func (r *OpenStackDataPlaneNodeReconciler) Reconcile(ctx context.Context, req ct
 		// update the overall status condition if service is ready
 		if instance.IsReady() {
 			instance.Status.Conditions.MarkTrue(condition.ReadyCondition, dataplanev1beta1.DataPlaneNodeReadyMessage)
-		}
-		c := instance.Status.Conditions.Mirror(condition.ReadyCondition)
-		if c.Reason == condition.ErrorReason {
-			instance.Status.Conditions.MarkFalse(
-				condition.ReadyCondition,
-				condition.ErrorReason,
-				condition.SeverityError,
-				c.Message)
+		} else {
+			// something is not ready so reset the Ready condition
+			instance.Status.Conditions.MarkUnknown(
+				condition.ReadyCondition, condition.InitReason, condition.ReadyInitMessage)
+			// and recalculate it based on the state of the rest of the conditions
+			instance.Status.Conditions.Set(instance.Status.Conditions.Mirror(condition.ReadyCondition))
 		}
 		err := helper.PatchInstance(ctx, instance)
 		if err != nil {

--- a/controllers/openstackdataplanerole_controller.go
+++ b/controllers/openstackdataplanerole_controller.go
@@ -142,14 +142,12 @@ func (r *OpenStackDataPlaneRoleReconciler) Reconcile(ctx context.Context, req ct
 		// update the overall status condition if service is ready
 		if instance.IsReady() {
 			instance.Status.Conditions.MarkTrue(condition.ReadyCondition, dataplanev1beta1.DataPlaneRoleReadyMessage)
-		}
-		c := instance.Status.Conditions.Mirror(condition.ReadyCondition)
-		if c.Reason == condition.ErrorReason {
-			instance.Status.Conditions.MarkFalse(
-				condition.ReadyCondition,
-				condition.ErrorReason,
-				condition.SeverityError,
-				c.Message)
+		} else {
+			// something is not ready so reset the Ready condition
+			instance.Status.Conditions.MarkUnknown(
+				condition.ReadyCondition, condition.InitReason, condition.ReadyInitMessage)
+			// and recalculate it based on the state of the rest of the conditions
+			instance.Status.Conditions.Set(instance.Status.Conditions.Mirror(condition.ReadyCondition))
 		}
 		err := helper.PatchInstance(ctx, instance)
 		if err != nil {

